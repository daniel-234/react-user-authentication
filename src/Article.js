import React from "react";

const Article = () => {
  return React.createElement("div", { className: "content" }, [
    React.createElement("h1", {}, "Welcome!"),
    React.createElement(
      "p",
      {},
      "This app is an exercise to demonstrate how to add authentication to a React app."
    )
  ]);
};

export default Article;
