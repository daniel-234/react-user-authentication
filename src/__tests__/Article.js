import React from "react";
import { create } from "react-test-renderer";
import Article from "../Article";

/*
 * Example of a snapshot test.
 */
test("Snapshot of the Article component", () => {
  /*
   * The `react-test-renderer` package can be used to
   * render React components to pure JavaScript objects,
   * without depending on the DOM.
   * The `create` method creates a `TestRenderer` instance
   * with the passed React element. It doesn't use the real
   * DOM, but it still fully renders the component tree into
   * memory, so you can make assertions about it.
   * See docs -> https://reactjs.org/docs/test-renderer.html
   */
  const article = create(<Article />);

  expect(article.toJSON()).toMatchSnapshot();
});
