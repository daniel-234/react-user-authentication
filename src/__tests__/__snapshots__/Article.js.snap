// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of the Article component 1`] = `
<div
  className="content"
>
  <h1>
    Welcome!
  </h1>
  <p>
    This app is an exercise to demonstrate how to add authentication to a React app.
  </p>
</div>
`;
